lib/rule_event_repository_serializer.rb

```diff
@@ -0,0 +1,13 @@
+class RuleEventRepositorySerializer
+  class << self
+    def dump(repository)
+      repository.dump
+    end
+
+    def load(data)
+      return RuleEventRepository.new if data.nil?
+
+      RuleEventRepository.load(data)
+    end
+  end
+end
```

Comment:
With the JSON database format, this is now much simpler.

One one hand, serialization != business logic. On the other hand, I could see `RuleEventRepositorySerializer` logic being folded in `RuleEventRepository` and to use `serialize :events, RuleEventRepository`.

While verbose, and since already coded, I think it make sense to keep them separate. If you prefer I can make the change. Since this is only ruby and not related to the data model, I don't really mind either way.